slam_toolbox/include/slam_toolbox/slam_toolbox_common.hpp

```diff
@@ -103,13 +104,14 @@ class SlamToolbox
   std::unique_ptr<tf2_ros::Buffer> tf_;
   std::unique_ptr<tf2_ros::TransformListener> tfL_;
   std::unique_ptr<tf2_ros::TransformBroadcaster> tfB_;
-  std::unique_ptr<message_filters::Subscriber<sensor_msgs::LaserScan> > scan_filter_sub_;
-  std::unique_ptr<tf2_ros::MessageFilter<sensor_msgs::LaserScan> > scan_filter_;
+  std::vector<std::unique_ptr<message_filters::Subscriber<sensor_msgs::LaserScan> > > scan_filter_subs_;
+  std::vector<std::unique_ptr<tf2_ros::MessageFilter<sensor_msgs::LaserScan> > > scan_filters_;
   ros::Publisher sst_, sstm_;
   ros::ServiceServer ssMap_, ssPauseMeasurements_, ssSerialize_, ssDesserialize_;
 
   // Storage for ROS parameters
-  std::string odom_frame_, map_frame_, base_frame_, map_name_, scan_topic_;
+  std::string map_frame_, map_name_;
+  std::vector<std::string> odom_frames_, base_frames_, laser_topics_;
```

Comment:
For things like these + the other maps and vectors you actually need to access the information from, it might be worth actually having a new `struct` which contains some of this information and then have a map of structs keyed to the frames so you can say `MyObj[base_frame]` contains a struct with the pose helper, laser assistent, transforms, etc. It might make things easier if there are functions that you access multiple of these maps to get an object from. 

If there is no single function where you access multiple of these resources, you can disregard.

Reply:
There are just alot of vector lookups of related information that might be well stored together. for example, these 3 frames might be better stored together where maybe one of the frames is keyed and the value is a struct containing the other 2 so that you can always get the relative important information all stored together

Reply:
I'd say that this change isn't necessary. The odom_frames_, base_frames_, and laser_topics_ are primarily used in functions that are run once eg. SlamToolbox constructor, setting up ROS interface, etc. I also looked into where some of the other data are used, and there isn't a common situation where some of the data is used together often, so grouping wouldn't simplify the code or improve performance